_posts/2024-06-01-galois.md

```diff
@@ -0,0 +1,60 @@
+and look at the determinant of this map, this is called the field map $$N_{L|K}: L \rightarrow k,\ \ a \mapsto N_{L|K}(a)$$.
+This map has many cool properties like being multiplicative (i.e. a homorphism of the unit groups) and if you consider an element $$a \in L$$ with $$m_a = X^n \cdot \alpha_n + ... + \alpha_0$$ then
+$$N_{L|K}(a) = \pm \alpha_0$$ the sign depending on the degree of the extension and the degree of the minimal polynomial. This was a step I had in mind when I created the challenge.
+Using this, and the fact that we have multiple instances, we can reduce the key space enough to iterate the remaining possibilities to find the flag.
+
+### Iterating the possible values
+Galios Theory tells us even more (this was the inital starting point for this challenge...): 
+$$N_{L|K}(a) = \Pi_{\sigma \in GAL(L|K)} \sigma(a)$$. To explain what galios groups are would go to far but for finite fields we know that they are generated by Frobenius homomorphisms. This means that they are generated by $$ x \mapsto x^p$$ with p being the characteristic. Thus the norm boils down to $$N_{L|K}(a) = a^\alpha$$ for the right $$\alpha$$ which can simply be calculated or just be looked up on e.g. wikipedia. Thinking about this a bit longer (you want to get yourself familiar with the concept of primitiv roots) we find out that solutions to equations of the from $$y = x^\alpha$$ can be altered by the unit group. More formally, if $$\omega$$ is a solution to the equation above and $$\phi^\alpha=1$$ then $$\omega \cdot \phi$$ is indeed also an solution. Personally I am not totally sure  about the number of "nontrivial" solutions (or any other detail), so you might want to look this  up for peace of mind.
+
+If we now go back to the challenge we get that for one instance the possible values for the flag form a line. If we took two instances and they would intersect we would have found the flag. Sadly they don't (again I don't remember that part of the solve that well so it is left as an exercise).
+But this essentially means we can combine multiple instances to get an  iterator that skips more elements. This construction is bascially a kind of chinese remainder theorem.
+My  solve script especially the last part is extremly ugly so I wont publish it, but I have included the challenge files somewhere so feel free to try to solve it yourself.
+
+### Related resources 
+If you want to know more about Galois theory, consider visiting a university course if you have the chance, otherwise there are many great online resources and books, such as [this](https://www.maths.ed.ac.uk/~tl/gt/gt.pdf) (be warned that most of Galois theory is not about finite fields), but you will probably want to start with an introduction to groups and algebra in general first.
+Somewhere here should be a link to the challenge files
+[Here](https://files.ctf.kitctf.de/trapdoor/a053f0f33932977546aa3e9720188e42cd8a1c6921ce95908f5f35766e2f53d6/trapdoor.tar.gz) the handout 
```

Comment:
This url will potentially be offline soonish™

Reply:
@intrigus-lgtm we should think about a solution for this when we archive the event page.

Reply:
Handout files should be added to the static github pages repository.
Only question would be what to do with sourceless challenges.
Without a handout and without a live instance they are unplayable :thinking: 
I think last year all challenges had a handout with the source (if we ignore the github actions challenge, which however has been preserved in another way), but this year that's not true.

Reply:
not my beer